In [2]:
import urllib.request
import json

# Request data goes here
# The example below assumes JSON formatting which may be updated
# depending on the format your endpoint expects.
# More information can be found here:
# https://docs.microsoft.com/azure/machine-learning/how-to-deploy-advanced-entry-script
data =  {
   "Inputs": {
     "input1":
       [
         { "PatientID": 1882185,
           "Pregnancies": 9,
           "PlasmaGlucose": 104,
           "DiastolicBloodPressure": 51,
           "TricepsThickness": 7,
           "SerumInsulin": 24,
           "BMI": 27.36983156,
           "DiabetesPedigree": 1.3504720469999998,
           "Age": 43 }
         ]
       },
 }

body = str.encode(json.dumps(data))

url = 'URL removed for security reasons'
# Replace this with the primary/secondary key, AMLToken, or Microsoft Entra ID token for the endpoint
api_key = 'API key removed for security reasons'
if not api_key:
    raise Exception("A key should be provided to invoke the endpoint")


headers = {'Content-Type':'application/json', 'Accept': 'application/json', 'Authorization':('Bearer '+ api_key)}

req = urllib.request.Request(url, body, headers)

try:
    response = urllib.request.urlopen(req)

    result = response.read()
    print(result)
except urllib.error.HTTPError as error:
    print("The request failed with status code: " + str(error.code))

    # Print the headers - they include the requert ID and the timestamp, which are useful for debugging the failure
    print(error.info())
    print(error.read().decode("utf8", 'ignore'))


b'{"Results": {"WebServiceOutput0": [{"PatientID": 1882185.0, "DiabetesPrediction": 1.0, "Probability": 0.7030538662687887}]}}'


The deployed Azure ML ACI endpoint was tested using a sample JSON request containing a single patient’s diagnostic data.
The model returned the following response:
    b'{"Results": {"WebServiceOutput0": [{"PatientID": 1882185.0, "DiabetesPrediction": 1.0, "Probability": 0.7030538662687887}]}}'
    
Interpretation of the Output
Field	-Meaning
PatientID	-Unique identifier of the patient from the input data
DiabetesPrediction	-1.0 indicates that the model predicts the patient has diabetes
Probability	-Confidence score (0–1). Here, the model is 70.3% confident in its prediction

Summary

The model successfully processed the input and predicted that:

Patient 1882185 is likely diabetic (Prediction = 1) with a probability score of 0.703.

This confirms that:

The inference pipeline works correctly

The ACI endpoint is functional

The feature transformations, scoring model, and custom Python output script are working end-to-end